packages/@react-spectrum/provider/src/Provider.tsx

```diff
@@ -146,6 +146,18 @@ const ProviderWrapper = React.forwardRef(function ProviderWrapper(props: Provide
     colorScheme: props.colorScheme ?? Object.keys(theme).filter(k => k === 'light' || k === 'dark').join(' ')
   };
 
+  let hasWarned = useRef(false);
+  useEffect(() => {
+    if (locale && domRef.current && process.env.NODE_ENV !== 'production') {
+      let closestLang = domRef.current.parentElement.closest('[lang]');
+      let lang = closestLang && closestLang.getAttribute('lang');
+      if (lang && lang !== locale && !hasWarned.current) {
+        console.warn(`Locales cannot be nested. ${locale} inside ${lang}.`);
```

Comment:
You'll need to update `hasWarned` here right?
```suggestion
        console.warn(`Locales cannot be nested. ${locale} inside ${lang}.`);
        hasWarned.current = true;
```